specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/DatabaseSqlVulnerabilityAssessmentRuleBaselines.json

```diff
@@ -0,0 +1,525 @@
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully got the sql vulnerability assessment rule baseline list.",
+            "schema": {
+              "$ref": "#/definitions/DatabaseSqlVulnerabilityAssessmentRuleBaselineListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 SqlVulnerabilityAssessmentIsDisabled - SQL vulnerability assessment feature is disabled. please enable the feature before executing other SQL vulnerability assessment operations.\n\n * 400 SqlVulnerabilityAssessmentScanResultsAreNotAvailableYet - SQL vulnerability assessment results are not available yet, please try again later.\n\n * 400 SqlVulnerabilityAssessmentInvalidRuleId - The SQL vulnerability assessment rule id is invalid.\n\n * 400 SqlVulnerabilityAssessmentScanDoesNotExist - SQL vulnerability assessment scan does not exist.\n\n * 400 SqlVulnerabilityAssessmentNoBaseline - No baseline have been found for the latest scan in the resource\n\n * 400 SqlVulnerabilityAssessmentNoRuleBaseline - No SQL vulnerability assessment baseline was found\n\n * 400 SqlVulnerabilityAssessmentBaselineNoScanResults - No scan results have been found for rule Id. To set a baseline there must be results for this rule in the latest scan available\n\n * 400 SqlVulnerabilityAssessmentBadBinaryRuleFormat - Input for binary rule is not a boolean representation\n\n * 400 SqlVulnerabilityAssessmentBadRuleFormat - The provided results do not comply with the actual layout of the scan results\n\n * 400 SqlVulnerabilityAssessmentBadRuleWithoutRuleIdFormat - The provided results do not comply with the actual layout of the scan results\n\n * 400 SqlVulnerabilityAssessmentBadBinaryRuleWithoutRuleIdFormat - Input for binary rule is not a boolean representation\n\n * 400 SqlVulnerabilityAssessmentBaselineNoScanResultsWithoutRuleId - No scan results have been found for one of the rules. To set a baseline there must be results for this rule in the latest scan available\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 SourceDatabaseNotFound - The source database does not exist.\n\n * 404 DatabaseDoesNotExist - User has specified a database name that does not exist on this server instance.\n\n * 409 DatabaseVulnerabilityAssessmentScanIsAlreadyInProgress - SQL Vulnerability Assessment scan is already in progress\n\n * 500 DatabaseIsUnavailable - Loading failed. Please try again later."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets a list of database's sql vulnerability assessment rule baselines.": {
+            "$ref": "./examples/DatabaseSqlVulnerabilityAssessmentRuleBaselineListByDatabase.json"
+          }
+        }
+      },
+      "post": {
```

Comment:
>"post" [](http://example.com/codeflow?start=6&length=6)

GET and POST? Not sure what the intention is here, doesn't seem correct, POST are reserved for actions only.

Reply:
post in this case is needed because it will discard any old existing baseline on that database and sets the baselines as specified exactly in this request. please let me know if that makes sense

Reply:
I am still unsure about your POST action, it seems like it is creating some sort of resource(s) internally. 

POSTs are not meant to create resources, POSTs are reserved for actions, see our documentation (first bullet point): 

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

You are modeling your API incorrectly then.

It seems to me that you are invoking an action to create a set of baselineRules? which is invalid since you are using POST. For example, what happens when a user calls your POST API multiple times, what happens with the previous rules?

To me it seems like the modeling needs to be the following (assuming you only want to support 1 baseline):

Collection GET -> `/baselines`

CRUD on `/baselines/default` -> default is used as a singleton, it means you are only supporting one baseline at a time - otherwise the API can model the API as `/baselines/{baselineName}` if you want to support multiple baselines.

Then the PUT body for the above API will be your Baseline input.

Now, to retrieve baseline rules:

Collection GET -> `/baselines/default/rules` to return all baseline rules
CRUD on `/baselines/default/rules/{ruleId}`

I strongly recommend reviewing our documentation: [https://armwiki.azurewebsites.net/rp_onboarding/process/onboarding.html](https://armwiki.azurewebsites.net/rp_onboarding/process/onboarding.html) and RPC: [https://github.com/Azure/azure-resource-manager-rpc](https://github.com/Azure/azure-resource-manager-rpc)